lib/pronto/github.rb

```diff
@@ -66,12 +71,11 @@ def create_commit_status(status)
     def create_pull_request_review(comments)
       options = {
         event: @config.github_review_type,
-        accept: 'application/vnd.github.v3.diff+json', # https://developer.github.com/v3/pulls/reviews/#create-a-pull-request-review
         comments: comments.map do |comment|
           {
-            path:     comment.path,
-            position: comment.position,
```

Comment:
Sorry, should've been more specific. I meant this invocation -- the other seems to have the fallback in place.

Reply:
This one does not need a fallback because the client directly accepts the options instead of using positional params.

Reply:
So, do all versions of `octokit` accept `line` instead of `position` for this API call?

Reply:
Yes. They all accept an options hash that is submitted as the POST body

https://github.com/octokit/octokit.rb/blob/v4.7.0/lib/octokit/client/reviews.rb#L96
https://github.com/octokit/octokit.rb/blob/v5.6.1/lib/octokit/client/reviews.rb#L92
https://github.com/octokit/octokit.rb/blob/v6.1.1/lib/octokit/client/reviews.rb#L92
https://github.com/octokit/octokit.rb/blob/v7.2.0/lib/octokit/client/reviews.rb#L92
https://github.com/octokit/octokit.rb/blob/v8.0.0/lib/octokit/client/reviews.rb#L92

Reply:
Cool, so the client just passes through to the [server endpoint](https://docs.github.com/en/rest/pulls/reviews?apiVersion=2022-11-28#create-a-review-for-a-pull-request) as-is, which supports both `line` and `position`.

However, I'm a bit confused by the description of the endpoint:
> position: The position in the diff where you want to add a review comment. **Note this value is not the same as the line number in the file**. For help finding the position value, read the note below.

Elsewhere, they say:
> Note: To comment on a specific line in a file, you need to first determine the position of that line in the diff. The GitHub REST API offers the application/vnd.github.v3.diff [media type](https://docs.github.com/rest/overview/media-types#commits-commit-comparison-and-pull-requests). To see a pull request diff, add this media type to the Accept header of a call to the [single pull request](https://docs.github.com/rest/pulls/pulls#get-a-pull-request) endpoint.

> The position value equals the number of lines down from the first "@@" hunk header in the file you want to add a comment. The line just below the "@@" line is position 1, the next line is position 2, and so on. The position in the diff continues to increase through lines of whitespace and additional hunks until the beginning of a new file.

So it looks like using `line` may be better in this case anyway, but I'm not sure I ever saw misplaced comments by pronto so @pbstriker38 would like your view on the interpretation as well before merging :+1:

Reply:
Yes, that note is a bit confusing but I think it might be a legacy thing from when the pull request comments API did not return line numbers.

I ran this branch on my fork of pronto and the comments were placed on the correct lines. I can point one of my repos at work to use this branch as well.

Reply:
### Here it is running on one of my repos

![image](https://github.com/prontolabs/pronto/assets/5264240/a3a2fa01-169d-44cb-b14b-ab7d339bbad1)

![image](https://github.com/prontolabs/pronto/assets/5264240/1f3eaeb2-3e6e-4f2f-9100-bfed05f0076a)

![image](https://github.com/prontolabs/pronto/assets/5264240/df7e2a95-b57c-4882-a6fe-0a0a825e618e)

![image](https://github.com/prontolabs/pronto/assets/5264240/f3a8c200-89a3-4e22-bbd0-f5549c9929b9)




### And then I fixed some offenses and created a new offense

![image](https://github.com/prontolabs/pronto/assets/5264240/2e55c467-0a2f-43f9-99ea-aae0d4627d1c)

![image](https://github.com/prontolabs/pronto/assets/5264240/99e9d3ce-56a1-468f-8e03-c6ac081f4e57)